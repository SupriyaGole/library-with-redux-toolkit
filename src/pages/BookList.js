import React from "react";
import {useSelector} from "react-redux";
import {SimpleGrid, Box, Text} from "@chakra-ui/react";

const BookList = () => {
  const {books} = useSelector((state) => state.book);
  return (<SimpleGrid
    h="100px"
    templateRows="repeat(2, 1fr)"
    templateColumns="repeat(5, 1fr)"
    gap={10}
    p={10}
  >
    {books.map(book =>
      <Box p="5" boxShadow="xl" rounded="md" bg="lightgray" color="black" key={book.title}>
        <Text fontSize="xl">{book.title}</Text>
        <Text fontSize="l">{book.author}</Text>
      </Box>)
    }
  </SimpleGrid>)
};

export default BookList;
