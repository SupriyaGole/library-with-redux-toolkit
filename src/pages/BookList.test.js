import React from "react";
import {render} from "@testing-library/react";
import BookList from "./BookList";
import {Provider} from "react-redux";
import { store } from '../redux/store';

describe("BookList", () => {
  test("should render book list with default books", () => {
    const {container, baseElement} = render(
      <Provider store={store}>
        <BookList/>
      </Provider>);
    expect(container).toMatchSnapshot();
    expect(baseElement).toMatchSnapshot();
  });
});
