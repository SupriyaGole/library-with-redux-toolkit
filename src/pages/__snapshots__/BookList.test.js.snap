// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList should render book list with default books 1`] = `
<div>
  <div
    class="css-wn1fcp"
  >
    <div
      class="css-1yugytp"
    >
      <p
        class="chakra-text css-6dvxm2"
      >
        1984
      </p>
      <p
        class="chakra-text css-16mom77"
      >
        George Orwell
      </p>
    </div>
    <div
      class="css-1yugytp"
    >
      <p
        class="chakra-text css-6dvxm2"
      >
        Harry Potter and the Philosopher's Stone
      </p>
      <p
        class="chakra-text css-16mom77"
      >
        J. K. Rowling
      </p>
    </div>
    <div
      class="css-1yugytp"
    >
      <p
        class="chakra-text css-6dvxm2"
      >
        The Lord of the Rings
      </p>
      <p
        class="chakra-text css-16mom77"
      >
        J.R.R Tolkien
      </p>
    </div>
  </div>
</div>
`;

exports[`BookList should render book list with default books 2`] = `
<body>
  <div
    id="chakra-toast-portal"
  >
    <ul
      id="chakra-toast-manager-top"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-top-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-top-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom-left"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
    <ul
      id="chakra-toast-manager-bottom"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
    />
    <ul
      id="chakra-toast-manager-bottom-right"
      style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
    />
  </div>
  <div>
    <div
      class="css-wn1fcp"
    >
      <div
        class="css-1yugytp"
      >
        <p
          class="chakra-text css-6dvxm2"
        >
          1984
        </p>
        <p
          class="chakra-text css-16mom77"
        >
          George Orwell
        </p>
      </div>
      <div
        class="css-1yugytp"
      >
        <p
          class="chakra-text css-6dvxm2"
        >
          Harry Potter and the Philosopher's Stone
        </p>
        <p
          class="chakra-text css-16mom77"
        >
          J. K. Rowling
        </p>
      </div>
      <div
        class="css-1yugytp"
      >
        <p
          class="chakra-text css-6dvxm2"
        >
          The Lord of the Rings
        </p>
        <p
          class="chakra-text css-16mom77"
        >
          J.R.R Tolkien
        </p>
      </div>
    </div>
  </div>
</body>
`;
